TEST.js

```diff
@@ -1,183 +0,0 @@
-//readline library link
-const readline = require("readline");
-const rl = readline.createInterface(process.stdin, process.stdout);
-
-function ask(questionText) {
-  return new Promise((resolve, reject) => {
-    rl.question(questionText, resolve);
-  });
-}
-
-start();
-//Variable Declarations
-let maxRange;
-let minRange;
-let numGuess;
-let compPick = "";
-
-//Game Start
-async function start() {
-  console.log(
-    "Let's play a game where you (Big BAUSE) make up a number and I (BeepBoopMachine) try to guess it. \n"
-  );
-  setInitRange();
-}
-
-//Set initial range with stops to make sure input is a number
-async function setInitRange() {
-  maxRange = await ask("So what's the highest number I can guess?\n");
-  while (isNaN(maxRange)) {
-    maxRange = await ask("Let's try this again. Please enter a number. \n");
-  }
-  maxRange = Number(maxRange);
-
-  minRange = await ask("And what is the lowest number I can guess?\n");
-  while (isNaN(minRange)) {
-    minRange = await ask("Let's try this again. Please enter a number.\n");
-  }
-  minRange = Number(minRange);
-
-  rangeCheck();
-}
-//Initial range check - make sure minRange < maxRange//
-async function rangeCheck() {
-  if (minRange > maxRange) {
-    console.log("It looks like your range makes no sense. Lets try it again\n");
-    setInitRange();
-  } else if (minRange === maxRange) {
-    console.log("Well it seems as if " + minRange + " is your number\n");
-    guessRightLoop();
-  } else if (minRange < maxRange) {
-    console.log(
-      `Okay cool, I can only guess numbers between ` +
-        minRange +
-        ` and ` +
-        maxRange +
-        `. Let us begin.`
-    );
-    compPickLoop();
-  }
-}
-
-//Random number picker within range (inclusive) //
-function pickNum(min, max) {
-  let range = max - min + 1;
-  let randInt = min + Math.floor(Math.random() * range);
-  return randInt;
-}
-
-//Smart number picker - mid point within range.
-function smartNum(min, max) {
-  let range = max - min + 1;
-  let midInt = Math.floor(range/2);
-  return midInt;
-}
-
-//Is numGuess your number?//
-async function compPickLoop() {
-  numGuess = pickNum(minRange, maxRange);
-  //numGuess = smartNum(minRange, maxRange);
-
-  compPick = await ask("Is " + numGuess + " your number?\n");
-  if (
-    compPick === "n" ||
-    compPick === "No" ||
-    compPick === "NO" ||
-    compPick === "no"
-  ) {
-    //console.log check block start//
-    console.log("minRange is " + minRange);
-    console.log("maxRange is " + maxRange);
-    //console.log check block end//
-    guessWrongLoop();
-  } else if (
-    compPick === "y" ||
-    compPick === "Yes" ||
-    compPick === "YES" ||
-    compPick === "yes"
-  ) {
```

Comment:
Good guard clauses here. This worked well to catch my attempts to break your game. One other way you could approach this that could save some time and typing (especially in larger projects coming up) would be to create a sanitize function to call on fresh user inputs as they are taken in. For example, you could `.trim().toUpperCase()` your input variable `compPick` right after it is taken in on line 81. This would save you from having to account for all upper and lower case variations of [ yes, no ] as opposed to typing each individual option out like lines 83 - 86.